central/platform/reprocessor/reprocessor_impl.go

```diff
@@ -0,0 +1,170 @@
+package reprocessor
+
+import (
+	"context"
+	"sync/atomic"
+
+	alertDS "github.com/stackrox/rox/central/alert/datastore"
+	alertutils "github.com/stackrox/rox/central/alert/utils"
+	deploymentDS "github.com/stackrox/rox/central/deployment/datastore"
+	platformmatcher "github.com/stackrox/rox/central/platform/matcher"
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/search"
+)
+
+const batchSize = 5000
+
+var (
+	log = logging.LoggerForModule()
+
+	reprocessorCtx = sac.WithAllAccess(context.Background())
+)
+
+type platformReprocessorImpl struct {
+	alertDatastore      alertDS.DataStore
+	deploymentDatastore deploymentDS.DataStore
+	platformMatcher     platformmatcher.PlatformMatcher
+
+	// isStarted will make sure only one reprocessing routine runs for an instance of reprocessor
+	isStarted atomic.Bool
+}
+
+func New(alertDatastore alertDS.DataStore,
+	deploymentDatastore deploymentDS.DataStore,
+	platformMatcher platformmatcher.PlatformMatcher) PlatformReprocessor {
+
+	return &platformReprocessorImpl{
+		alertDatastore:      alertDatastore,
+		deploymentDatastore: deploymentDatastore,
+		platformMatcher:     platformMatcher,
+	}
+}
+
+func (pr *platformReprocessorImpl) Start() {
+	swapped := pr.isStarted.CompareAndSwap(false, true)
+	if !swapped {
+		log.Error("Platform reprocessor already running")
+		return
+	}
+	go pr.runReprocessing()
```

Comment:
This is not cancelable, this means that if a user wants to cancel the central pod the termination may take a long time until the queries have finished. (Looking at huge alerting tables)

Reply:
Yeah that makes sense. I will make it cancelable.

Reply:
Done